### PR TITLE
Renaming earth to earthly

### PR DESCRIPTION
**Pull request by alexcb**
- when a user installs via brew, they type `brew install earthly`, so one would expect to find a command named `earthly` rather than `earth`
 - this git repo is located at `github.com/earthly/earthly.git` and not `github.com/earthly/earth.git`
 - Googling for `earthly` manages to find our documentation, where as the search-space for `earth` is much more crowded
 - `earth` easily tabs completes into `earthly`

### CODE REVIEW

File: earthly [hunk 1]
```diff
@@ -1,12 +1,12 @@
 #!/bin/bash
 set -e
 
-bindir="$HOME/.earthly"
+bindir="$HOME/.earth"
```
Comment (vladaionescu) [inline earthly:4]: Why not use `.earthly` for this too?

File: buildcontext/excludes.go [hunk 2]
```diff
@@ -8,12 +8,12 @@ import (
 	"github.com/pkg/errors"
 )
 
-// EarthIgnoreFile is the name of the earth ignore file.
+// EarthIgnoreFile is the name of the earthly ignore file.
 const EarthIgnoreFile = ".earthignore"
```
Comment (alexcb) [inline buildcontext/excludes.go:12]: we're going to need some backwards compatibility if we want to rename this to `.earthlyignore`

File: README.md [hunk 3]
```diff
@@ -319,7 +319,7 @@ release:
 ```
 
 ```bash
-earth --secret GITHUB_TOKEN --push +release
+earthly --secret GITHUB_TOKEN --push +release
```
Comment (vladaionescu) [inline README.md:322]: Make changes in README on `next` branch instead?

File: README.md [hunk 4]
```diff
@@ -145,8 +145,8 @@ See also the [full documentation](https://docs.earthly.dev).
 Reference pages
 
 * 📑 [Earthfile reference](https://docs.earthly.dev/earthfile)
-* #️⃣ [Earth command reference](https://docs.earthly.dev/earth-command)
-* ⚙️ [Configuration reference](https://docs.earthly.dev/earth-config)
+* #️⃣ [Earth command reference](https://docs.earthly.dev/earthly-command)
+* ⚙️ [Configuration reference](https://docs.earthly.dev/earthly-config)
```
Comment (vladaionescu) [inline README.md:149]: Note that these will require redirects in gitbook. See https://docs.gitbook.com/integrations/github/content-configuration

**Review by vladaionescu — COMMENTED** [review]
* Also rename `docker2earth`?
* `earthApp` -> `earthlyApp` in main.go
* Some other (maybe not very important) variables names still have names like `earthTargetName` (eg `artifact.go`). I guess it's ok to skip them for now.
* Docs will be done in separate PR branched off `next`, right?
* Changes to add symlink via bootstrap will be done in separate PR, right?

File: buildcontext/excludes.go (hunk 2 above)
Comment (vladaionescu) [reply buildcontext/excludes.go:12]: The other possible one is `Earthfile`... Perhaps let's leave them like this for now.

File: README.md (hunk 3 above)
Comment (alexcb) [reply README.md:322]: that's a good call, I left out the docs renaming for a different PR but let the README changes slip in, i'll save these for a future PR.

File: earthly (hunk 1 above)
Comment (alexcb) [reply earthly:4]: good catch, this was a mistaken search/replace.

**Comment by alexcb** [comment]
I renamed earthApp to earthlyApp in this PR; I'm hoping to do a follow up PR for docs, readmes, and other variable renames and more significant code changes like the bootstraping logic.

**Review by vladaionescu — APPROVED** [review]

**Pull request merged**